core/modules/keyboard.js

```diff
@@ -274,6 +273,80 @@ KeyboardManager.prototype.checkKeyDescriptors = function(event,keyInfoArray) {
 	return false;
 };
 
+KeyboardManager.prototype.getShortcutTiddlerList = function() {
+	return $tw.wiki.getTiddlersWithTag("$:/tags/KeyboardShortcut");
+};
+
+KeyboardManager.prototype.updateShortcutLists = function(tiddlerList) {
+	this.shortcutTiddlers = tiddlerList;
+	for(var i=0; i < tiddlerList.length; i++) {
+		var title = tiddlerList[i],
+			tiddlerFields = $tw.wiki.getTiddler(title).fields;
+		this.shortcutKeysList[i] = tiddlerFields.key !== undefined ? tiddlerFields.key : undefined;
+		this.shortcutActionList[i] = tiddlerFields.text;
+		this.shortcutParsedList[i] = this.shortcutKeysList[i] !== undefined ? this.parseKeyDescriptors(this.shortcutKeysList[i]) : undefined;
+	}
+};
+
+KeyboardManager.prototype.handleKeydownEvent = function(event) {
+	var key, action;
+	for(var i=0; i < this.shortcutTiddlers.length; i++) {
+		if(this.shortcutParsedList[i] !== undefined && this.checkKeyDescriptors(event,this.shortcutParsedList[i])) {
+			key = this.shortcutParsedList[i];
+			action = this.shortcutActionList[i];
+		}
+	}
+	if(key !== undefined) {
+		event.preventDefault();
+		event.stopPropagation();
+		$tw.rootWidget.invokeActionString(action,$tw.rootWidget);
+		return true;
+	}
+	return false;
+};
+
+// Returns true if a passed array contains a Tiddler that has changed
+KeyboardManager.prototype.hasAnyTiddlerChanged = function(changedTiddlers,tiddlerList) {
```

Comment:
I wonder if this helper might be better expressed as a description of what it does, rather than it's purpose here. If "hasOwnProperty" gave us `$tw.utils.hop`, perhaps this hasAtLeastOneOfTheseOwnProperties would give us `$tw.utils.halootop` Or maybe just `$tw.utils.hopArray` would be more sensible.

Reply:
should this `$tw.utils.hopArray` go into boot.js?

Reply:
should this `$tw.utils.hopArray` go into boot.js?

Reply:
should this `$tw.utils.hopArray` go into boot.js?

Reply:
Sorry @BurningTreeC - we shouldn't put it in boot.js unless it's used in boot.js, for post-boot utils better to put it in `core/modules/utils/utils.js`

Reply:
@Jermolene done